infra/base-images/base-runner/dataflow_tracer.py

```diff
@@ -0,0 +1,143 @@
+#!/usr/bin/env python3
+# Copyright 2020 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+################################################################################
+"""Script for collecting dataflow traces using DFSan compiled binary. The script
+imitates `CollectDataFlow` function from libFuzzer but provides some flexibility
+for skipping long and/or slow corpus elements.
+
+Follow https://github.com/google/oss-fuzz/issues/1632 for more details."""
+import hashlib
+import os
+import subprocess
+import sys
+
+# These can be controlled by the runner in order to change the values without
+# rebuiding OSS-Fuzz base images.
+FILE_SIZE_LIMIT = int(os.getenv('DFT_FILE_SIZE_LIMIT', 32 * 1024))
```

Comment:
Just curious how you arrived at this default limit?

Reply:
I manually looked at the size of the corpus elements for libexif and libyaml, and then played with few different limits for this: https://github.com/google/oss-fuzz/issues/1632#issuecomment-575344059

It seemed like 95+% of corpora fits this limit and increasing it even more doesn't increase the success ratio drastically. Also important point here is that dataflow trace is being collected for every 16 bytes of the input, i.e. 32KB inputs would require 2K executions.

I consider this limit rather conservative, to make sure the builds would finish fast enough.